samples/cpp/bgfg_segm.cpp

```diff
@@ -1,111 +1,113 @@
 #include "opencv2/core.hpp"
-#include <opencv2/core/utility.hpp>
 #include "opencv2/imgproc.hpp"
-#include "opencv2/video/background_segm.hpp"
+#include "opencv2/video.hpp"
 #include "opencv2/videoio.hpp"
 #include "opencv2/highgui.hpp"
-#include <stdio.h>
+#include <iostream>
 
 using namespace std;
 using namespace cv;
 
-static void help()
-{
- printf("\nDo background segmentation, especially demonstrating the use of cvUpdateBGStatModel().\n"
-"Learns the background at the start and then segments.\n"
-"Learning is togged by the space key. Will read from file or camera\n"
-"Usage: \n"
-"			./bgfg_segm [--camera]=<use camera, if this key is present>, [--file_name]=<path to movie file> \n\n");
-}
-
-const char* keys =
-{
-    "{c  camera   |         | use camera or not}"
-    "{m  method   |mog2     | method (knn or mog2) }"
-    "{s  smooth   |         | smooth the mask }"
-    "{fn file_name|../data/tree.avi | movie file        }"
-};
-
-//this is a sample for foreground detection functions
 int main(int argc, const char** argv)
 {
-    help();
-
+    const String keys = "{c camera||use video stream from camera (default is NO)}"
+                        "{fn file_name|../data/tree.avi|video file}"
+                        "{m method|mog2|method: background subtraction algorithm ('knn', 'mog2')}"
+                        "{h help||show help message}";
     CommandLineParser parser(argc, argv, keys);
+    parser.about("This sample demonstrates background segmentation.");
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
     bool useCamera = parser.has("camera");
-    bool smoothMask = parser.has("smooth");
-    string file = parser.get<string>("file_name");
-    string method = parser.get<string>("method");
-    VideoCapture cap;
-    bool update_bg_model = true;
-
-    if( useCamera )
-        cap.open(0);
-    else
-        cap.open(file.c_str());
-
-    parser.printMessage();
-
-    if( !cap.isOpened() )
+    String file = parser.get<String>("file_name");
+    String method = parser.get<String>("method");
+    if (!parser.check())
     {
-        printf("can not open camera or video file\n");
-        return -1;
+        parser.printErrors();
+        return 1;
     }
 
-    namedWindow("image", WINDOW_NORMAL);
-    namedWindow("foreground mask", WINDOW_NORMAL);
-    namedWindow("foreground image", WINDOW_NORMAL);
-    namedWindow("mean background image", WINDOW_NORMAL);
+    VideoCapture cap(useCamera ? 0 : file.c_str());
```

Comment:
> useCamera ? 0 : file.c_str()

This calls "VideoCapture((const char*)NULL)" instead of "VideoCapture((int)0)".
Consider using `.open()` method of `cv::Ptr<VideoCapture>` with different initializers.

Reply:
Fixed, thank you!